File: spark-study-sql-scala/src/main/java/com/wen/spark/sql/core/ParquetLoadData.scala
package com.wen.spark.sql.core

import org.apache.spark.SparkConf
import org.apache.spark.api.java.JavaSparkContext
import org.apache.spark.api.java.function.Function
import org.apache.spark.sql._

import scala.collection.immutable.List
object ParquetLoadData {
  def main(args: Array[String]): Unit = {
    val conf = new SparkConf().setAppName("ParquetLoadData")
    val sc = new JavaSparkContext(conf)
    val sqlContext = new SQLContext(sc)
    val reader = sqlContext.read
    val ds = reader.json("hdfs://hadoop:8020/data/users.parquet")
    ds.show()
    ds.registerTempTable("users")
    val userName = sqlContext.sql("select name from users")
    //RDD<ROW>
    import scala.collection.JavaConversions._
    //对查询出来的dataSet 进行操作，处理打印
  val userNameRDD = userName.javaRDD.map(new Function[Row, String]() {
    @throws[Exception]
    override def call(o: Row): String = {
      return "name:" + o.getString(0)
    }
  }).collect()

    for (usernmae <- userNameRDD) {
      println(usernmae)
    }
  }
}

File: spark-study-sql-java/src/main/java/com/wen/spark/sql/core/ParquetLoadData.java
package com.wen.spark.sql.core;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.api.java.function.Function;
import org.apache.spark.sql.DataFrameReader;
import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.SQLContext;
import java.util.List;
import org.apache.spark.sql.Row;

public class ParquetLoadData {
    public static void main(String[] args) {
        SparkConf conf=new SparkConf().setAppName("ParquetLoadData").setMaster("local");
        JavaSparkContext sc=new JavaSparkContext(conf);
        System.setProperty("HADOOP_USER_NAME", "root");
        sc.hadoopConfiguration().set("dfs.client.use.datanode.hostname","true");
        SQLContext sqlContext=new SQLContext(sc);
        DataFrameReader reader=sqlContext.read();
        Dataset ds= reader.load("hdfs://cloud.codeguoj.cn:8020/test/parquet/");
        ds.show();
        ds.registerTempTable("users");
        Dataset userName=sqlContext.sql("select name from users");
        //对查询出来的dataSet 进行操作，处理打印
        List<String> userNameRDD=userName.javaRDD().map(new Function<Row,String>() {
            public String call(Row o) throws Exception {
                return "name:"+o.getString(0);
            }
        }).collect();
        for(String usernmae:userNameRDD){
            System.out.println(usernmae);
        }
    }
}
